twilio-authenticator-sample/src/main/java/com/twilio/authenticatorsample/approvalrequests/RequestsFragment.java

```diff
@@ -54,18 +54,20 @@
     private MessageHelper messageHelper;
 
     private ApprovalRequests approvalRequests;
-
-    public RequestsFragment() {
-        // Required empty public constructor
-    }
+    private Long appId;
 
     /**
      * Use this factory method to create a new instance of Requests fragment
      *
      * @return A new instance of fragment RequestsFragment.
      */
-    public static RequestsFragment newInstance() {
-        return new RequestsFragment();
+    public static RequestsFragment newInstance(Long appId, TwilioAuthenticator twilioAuthenticator) {
+
+        RequestsFragment requestsFragment = new RequestsFragment();
+        requestsFragment.appId = appId;
```

Comment:
If we do this. The appId will be lost when the app goes to the background and the system tries to recreate the fragment. Please use:

``` 
Bundle arguments = new Bundle();
args.putLong("appId", appId);
requestFragment.setArguments(arguments);
```

You can retrieve this later with:

appId = getIntent().getLongExtra("appId", 0);